tests/plugins/test_history_at_collections.py

```diff
@@ -0,0 +1,355 @@
+        record2_deletion_last_modified = resp.json['data']['last_modified']
+
+        # Check view at 1, 2 and 3
+        bucket_records_uri = self.build_version_object_uri(self.collection_uri + '/records')
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri, record2_creation_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record2, self.record]
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri, record3_creation_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record3, record2, self.record]
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri, record2_deletion_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record3, self.record]
+
+    def test_we_can_get_a_bucket_record_list_at_a_certain_time_with_updated(self):
+        # Create a second record
+        record_2_uri = self.collection_uri + '/records/record2'
+        resp = self.app.put_json(record_2_uri,
+                                 {"data": {"foo": ["record2"]}},
+                                 headers=self.headers)
+        record2 = resp.json['data']
+        record2_creation_last_modified = record2['last_modified']
+
+        # Create a third record
+        record_3_uri = self.collection_uri + '/records/record3'
+        resp = self.app.put_json(record_3_uri,
+                                 {"data": {"foo": ["record3"]}},
+                                 headers=self.headers)
+        record3 = resp.json['data']
+        record3_creation_last_modified = record3['last_modified']
+
+        # Update the first record
+        resp = self.app.patch_json(self.record_uri,
+                                   {"data": {"foo": ['record1']}},
+                                   headers=self.headers)
+        record = resp.json['data']
+        record_update_last_modified = record['last_modified']
+
+        bucket_records_uri = self.build_version_object_uri(self.collection_uri + '/records')
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri, record2_creation_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record2, self.record]
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri, record3_creation_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record3, record2, self.record]
+
+        resp = self.app.get('{}/{}'.format(bucket_records_uri,
+                                           record_update_last_modified),
+                            headers=self.headers)
+        records = resp.json['data']
+        assert records == [record, record3, record2]
```

Comment:
Those tests look like huge copy-paste of the same steps. Plus the test methods are long and contains several steps. It's usually easier to have short well-scoped test methods